pkg/api/steve/supportconfigs/generator.go

```diff
@@ -0,0 +1,169 @@
+package supportconfigs
+
+import (
+	"archive/tar"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	v1 "github.com/rancher/rancher/pkg/generated/norman/core/v1"
+	"github.com/rancher/rancher/pkg/types/config"
+	"github.com/sirupsen/logrus"
+	authzv1 "k8s.io/api/authorization/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	authv1 "k8s.io/client-go/kubernetes/typed/authorization/v1"
+)
+
+const (
+	// Endpoint The endpoint that this URL is accessible at - used for routing and for SARs
+	Endpoint             = "/v1/generateSUSERancherSupportConfig"
+	cspAdapterNamespace  = "cattle-csp-system"
+	cspAdapterConfigmap  = "csp-config"
+	tarContentType       = "application/x-tar"
+	unauthenticatedError = "unauthenticaed"
+	forbiddenError       = "forbidden"
+)
+
+type GeneratorHandler struct {
+	ConfigMaps           v1.ConfigMapInterface
+	SubjectAccessReviews authv1.SubjectAccessReviewInterface
+	ConfigHandler        supportConfigHandler
+}
+
+func NewGeneratorHandler(scaledContext *config.ScaledContext) GeneratorHandler {
+	return GeneratorHandler{
+		ConfigMaps:           scaledContext.Core.ConfigMaps(cspAdapterNamespace),
+		SubjectAccessReviews: scaledContext.K8sClient.AuthorizationV1().SubjectAccessReviews(),
+		ConfigHandler:        supportConfigHandler{},
+	}
+}
+
+func (h *GeneratorHandler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+	err := h.authorize(request)
+	if err != nil {
+		switch err.Error() {
+		case unauthenticatedError:
+			writer.WriteHeader(http.StatusUnauthorized)
+			_, _ = writer.Write([]byte(err.Error()))
+		case forbiddenError:
+			writer.WriteHeader(http.StatusForbidden)
+			_, _ = writer.Write([]byte(err.Error()))
+		default:
+			writer.WriteHeader(http.StatusInternalServerError)
+			logrus.Errorf("Failed to authorize user with error: %s", err.Error())
+		}
+		return
+	}
+	logrus.Infof("Generating supportconfig")
+	archive, err := h.generateSupportConfig()
+	logrus.Infof("Done Generating supportconfig")
+	if err != nil {
+		logrus.Errorf("Error when generating support config %v", err)
+		writer.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	writer.Header().Set("Content-Type", tarContentType)
+	writer.Header().Set("Content-Disposition", "filename=\"supportConfig.tar\"")
```

Comment:
Does this need to include `attachment;`? https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition#filename